callouts/java/service-callout/src/main/java/service/ServiceCallout.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (c) 2024 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package service;
+
+import io.envoyproxy.envoy.service.ext_proc.v3.ExternalProcessorGrpc;
+import io.envoyproxy.envoy.service.ext_proc.v3.HttpBody;
+import io.envoyproxy.envoy.service.ext_proc.v3.HttpHeaders;
+import io.envoyproxy.envoy.service.ext_proc.v3.ProcessingRequest;
+import io.envoyproxy.envoy.service.ext_proc.v3.ProcessingResponse;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.netty.NettyServerBuilder;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.security.GeneralSecurityException;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpExchange;
+
+import static utils.SslUtils.createSslContext;
+import static utils.SslUtils.readFileToBytes;
+
+/**
+ * ServiceCallout provides a base class for handling HTTP request and response processing
+ * in a gRPC-based service callout server. It processes incoming requests and modifies headers,
+ * bodies, or immediate responses as per the service logic.
+ */
+public class ServiceCallout {
+    private static final Logger logger = Logger.getLogger(ServiceCallout.class.getName());
+
+    private Server server;
```

Comment:
I just caught this but we need two servers, the secure and plaintext ones.
i.e. `plaintextServer`